### PR TITLE
Update to latest SDK snapshot.

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/trace/TracingContextUtils.java [hunk 1]
```diff
@@ -77,22 +74,4 @@ public static Span getSpanWithoutDefault(
         io.opentelemetry.trace.Span.fromContextOrNull(agentContext);
     return agentSpan == null ? null : toApplication(agentSpan);
   }
-
-  public static Scope currentContextWith(Span applicationSpan) {
```
Comment (anuraaga) [inline instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/trace/TracingContextUtils.java:81]: I just deleted this since I think even previously, this wasn't required because of instrumentation of withSpan and makeCurrent

File: instrumentation/external-annotations/external-annotations.gradle [hunk 2]
```diff
@@ -1,5 +1,7 @@
 apply from: "$rootDir/gradle/instrumentation.gradle"
 
+// TODO(anuraaga): Reenable after SDK 0.10.0 is published
```
Comment (anuraaga) [inline instrumentation/external-annotations/external-annotations.gradle:3]: I guess this is better than having muzzle pull in random versions from snapshot repo? Let me know if there's a better way.

I'm surprised the smoke tests pass though - actually I had flaky behavior where sometimes they fail with no WithSpan and sometimes they pass fine locally, and have similar seemingly bad-cache behavior before so a bit anxious they may have some problem @iNikem

**Review by trask — APPROVED** [review]
Thanks for keeping us up-to-date!

File: instrumentation/external-annotations/external-annotations.gradle (hunk 2 above)
Comment (trask) [reply instrumentation/external-annotations/external-annotations.gradle:3]: Did something change in the annotations that broke muzzle's compatibility check?

Comment (anuraaga) [reply instrumentation/external-annotations/external-annotations.gradle:3]: Now the implementation uses `Span.makeCurrent()` instead of `TracingContextUtils` which isn't available in previous opentelemetry API

File: smoke-tests/grpc/build.gradle [hunk 3]
```diff
@@ -17,8 +24,8 @@ dependencies {
   implementation "io.grpc:grpc-netty-shaded:1.32.1"
   implementation "io.grpc:grpc-protobuf:1.32.1"
   implementation "io.grpc:grpc-stub:1.32.1"
-  implementation "io.opentelemetry:opentelemetry-proto:0.9.1"
-  implementation "io.opentelemetry:opentelemetry-extension-auto-annotations:0.9.1"
+  implementation "io.opentelemetry:opentelemetry-proto:0.10.0-SNAPSHOT"
+  implementation "io.opentelemetry:opentelemetry-extension-auto-annotations:0.10.0-SNAPSHOT"
```
Comment (trask) [inline smoke-tests/grpc/build.gradle:28]: is this change needed? smoke tests are passing on this PR, and I think(?) they use docker images from last build on master

File: smoke-tests/springboot/build.gradle [hunk 4]
```diff
@@ -11,11 +11,18 @@ version = '0.0.1-SNAPSHOT'
...
+    content {
+      includeGroup "io.opentelemetry"
+    }
+  }
 }
 
 dependencies {
   implementation 'org.springframework.boot:spring-boot-starter-web'
-  implementation 'io.opentelemetry:opentelemetry-extension-auto-annotations:0.7.0'
+  implementation 'io.opentelemetry:opentelemetry-extension-auto-annotations:0.10.0-SNAPSHOT'
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply smoke-tests/springboot/build.gradle:25]: Ah thanks I got confused since muzzle was failing

**Comment by trask** [comment]
just checking, did u want to keep the snapshot repository added in the 2 smoke test apps?

File: smoke-tests/springboot/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (anuraaga) [reply smoke-tests/springboot/build.gradle]: Ah wait - yeah the main point of the change wasn't the annotations but to pull in the newer API package (figured it's more idiomatic to just use the transitive dependency to manage that). I'm a bit confused why the smoke tests are passing - wouldn't you expect the instrumentation to get muzzled?

**Comment by anuraaga** [comment]
Nope :-) But realized I'm still confused on the version issue https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/1479#discussion_r512339537

File: smoke-tests/springboot/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (trask) [reply smoke-tests/springboot/build.gradle]: `WithSpanAnnotationInstrumentation` uses shaded `Span.makeCurrent`, but doesn't rely on OpenTelemetry API in user class loader, so no muzzle problem (though we want to apply muzzle consistently across all instrumentation in a single module in the future https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1274#issuecomment-707860035)

Oh, and this made me realized `@WithSpan` instrumentation is no longer in `external-annotations`, which makes me think maybe your change to `external-annotations.gradle` is not needed?

[the remark this answers is not in the document]
Comment (anuraaga) [reply smoke-tests/springboot/build.gradle]: Hmm - the `:muzzle` task was definitely failing in external-annotations though. Muzzle explicitly depends on the otel api, but should it be checking our shaded version instead?

[the remark this answers is not in the document]
Comment (trask) [reply smoke-tests/springboot/build.gradle]: oh sorry, we should remove the annotations dep from that muzzle block in `external-annotations.gradle` since that module doesn't use it anymore (we should have removed it when we moved WithSpan annotation instrumentation from `external-annotations` to `opentelemetry-api-beta`), e.g.

```
muzzle {
  pass {
    group = "io.opentracing.contrib.dropwizard"
    module = "dropwizard-opentracing"
    versions = "[0.2.2,)"
  }
}
```

[the remark this answers is not in the document]
Comment (anuraaga) [reply smoke-tests/springboot/build.gradle]: Thanks!

Not sure if you meant this literally or figuratively - I tried adding muzzle for other libraries as well but dropwizard-opentracing seems to be the only one that passes for some reason.

[the remark this answers is not in the document]
Comment (trask) [reply smoke-tests/springboot/build.gradle]: > Not sure if you meant this literally or figuratively

I use e.g. to mean a lot of things at once 😂
